src/core/qgsvectorlayer.cpp

```diff
@@ -2387,6 +2387,19 @@ long QgsVectorLayer::featureCount() const
          ( mEditBuffer ? mEditBuffer->mAddedFeatures.size() - mEditBuffer->mDeletedFeatureIds.size() : 0 );
 }
 
+long QgsVectorLayer::featureCount( QgsFeatureRequest request )
+{
+  QgsFeatureIterator fit = getFeatures( request );
```

Comment:
Hi @ahuarte47, the idea is not to count all the features, it's to count the features corresponding to a featureRequest probably made with an extent or fids or an expression.